x-pack/legacy/plugins/siem/public/containers/network/filter.tsx

```diff
@@ -61,90 +59,84 @@ interface NetworkFilterDispatchProps {
 
 export type NetworkFilterProps = OwnProps & NetworkFilterReduxProps & NetworkFilterDispatchProps;
 
-class NetworkFilterComponent extends React.PureComponent<NetworkFilterProps> {
-  private memoizedApplyFilterQueryFromKueryExpression: (expression: string) => void;
-  private memoizedSetFilterQueryDraftFromKueryExpression: (expression: string) => void;
+const NetworkFilterComponent = React.memo<NetworkFilterProps>(
+  ({
+    applyNetworkFilterQuery,
+    children,
+    indexPattern,
+    isNetworkFilterQueryDraftValid,
+    kueryFilterQuery,
+    networkFilterQueryDraft,
+    setNetworkFilterQueryDraft,
+    setQuery,
+    type,
+  }) => {
+    const applyFilterQueryFromKueryExpression = (expression: string) =>
+      applyNetworkFilterQuery({
+        filterQuery: {
+          kuery: {
+            kind: 'kuery',
+            expression,
+          },
+          serializedQuery: convertKueryToElasticSearchQuery(expression, indexPattern),
+        },
+        networkType: type,
+      });
 
-  constructor(props: NetworkFilterProps) {
-    super(props);
-    this.memoizedApplyFilterQueryFromKueryExpression = memoizeOne(
-      this.applyFilterQueryFromKueryExpression
+    const setFilterQueryDraftFromKueryExpression = (expression: string) =>
+      setNetworkFilterQueryDraft({
+        filterQueryDraft: {
+          kind: 'kuery',
+          expression,
+        },
+        networkType: type,
+      });
+    const refApplyFilterQueryFromKueryExpression = useRef<(expression: string) => void>(
+      applyFilterQueryFromKueryExpression
     );
-    this.memoizedSetFilterQueryDraftFromKueryExpression = memoizeOne(
-      this.setFilterQueryDraftFromKueryExpression
+    const refSetFilterQueryDraftFromKueryExpression = useRef<(expression: string) => void>(
```

Comment:
let's do the same technique that we are going to do for `x-pack/legacy/plugins/siem/public/containers/hosts/filter.tsx`